app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreNotifiers.tsx

```diff
@@ -0,0 +1,150 @@
+import { FC, useCallback, useEffect, useMemo, useState } from "preact/compat";
+import { useLocation } from "react-router";
+import { useNotifiersSetQueryParams as useSetQueryParams } from "./hooks/useSetQueryParams";
+import Spinner from "../../components/Main/Spinner/Spinner";
+import Alert from "../../components/Main/Alert/Alert";
+import Accordion from "../../components/Main/Accordion/Accordion";
+import { useFetchNotifiers } from "./hooks/useFetchNotifiers";
+import "./style.scss";
+import NotifiersHeader from "../../components/ExploreAlerts/NotifiersHeader";
+import NotifierHeader from "../../components/ExploreAlerts/NotifierHeader";
+import Notifier from "../../components/ExploreAlerts/Notifier";
+import { Notifier as APINotifier, Target as APITarget } from "../../types";
+import { getQueryStringValue } from "../../utils/query-string";
+import { getChanges } from "./helpers";
+
+const defaultKindsStr = getQueryStringValue("kinds", "") as string;
+const defaultKinds = defaultKindsStr.split("&").filter((rt) => rt) as string[];
+const defaultSearchInput = getQueryStringValue("search", "") as string;
+
+const ExploreNotifiers: FC = () => {
+  const {
+    notifiers,
+    isLoading: loadingNotifiers,
+    error: errorNotifiers,
+  } = useFetchNotifiers();
+
+  const [expandNotifiers, setExpandNotifiers] = useState<Set<string>>(
+    new Set<string>(),
+  );
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [kinds, setKinds] = useState(defaultKinds);
+
+  useSetQueryParams({
+    kinds: kinds.join("&"),
+    search: searchInput,
+  });
+
+  const { hash } = useLocation();
+
+  const isLoading = useMemo(() => {
+    return loadingNotifiers;
+  }, [loadingNotifiers]);
+
+  const error = useMemo(() => {
+    return errorNotifiers;
+  }, [errorNotifiers]);
+
+  useEffect(() => {
+    if (isLoading || !notifiers || !hash || error) return;
+
+    const target = document.querySelector(hash);
+    if (!target) return;
+
+    let parent = target.closest("details");
+    while (parent) {
+      parent.open = true;
+      parent = parent!.parentElement!.closest("details");
```

Comment:
I would suggest to add additional check for existence of `parentElement` and `closes` func (instead of `!.`) to avoid runtime error.